crates/wasi-common/winx/src/file.rs

```diff
@@ -57,8 +56,8 @@ impl FileType {
 
 pub unsafe fn get_file_type(handle: RawHandle) -> Result<FileType> {
     let file_type = FileType(GetFileType(handle));
-    let err = winerror::WinError::last();
-    if file_type.is_unknown() && err != winerror::WinError::ERROR_SUCCESS {
+    let err = Error::last_os_error();
+    if file_type.is_unknown() && err.raw_os_error().unwrap() as u32 != winerror::ERROR_SUCCESS {
```

Comment:
How about adding a comment why unwrap will always succeed (possibly using expect)?

Reply:
See discussion in #1242, plus [raw_os_error docs](https://doc.rust-lang.org/std/io/struct.Error.html#method.raw_os_error).